File: apps/web/src/types/attachment.ts
export interface Attachment {
  id: string;
  filename: string;
  content_type: string;
  size_bytes: number;
  scan_status: string;
  scan_status_label_da: string;
  scanned_at: string | null;
  created_at: string;
  download_available: boolean;
  file_retrievable: boolean;
  file_unavailable_label_da: string | null;
  can_delete: boolean;
}

File: apps/api/src/star_itsm_api/schemas/attachment.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class AttachmentRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    filename: str
    content_type: str
    size_bytes: int
    scan_status: str
    scan_status_label_da: str
    scanned_at: datetime | None
    created_at: datetime
    download_available: bool
    file_retrievable: bool
    file_unavailable_label_da: str | None = None
